miniapps/nurbs/nurbs_ex1.cpp

```diff
@@ -401,6 +405,34 @@ int main(int argc, char *argv[])
       sol_sock << "solution\n" << *mesh << x << flush;
    }
 
+   if (mesh->Dimension() == 1 && lod > 0)
+   {
+      ofstream sol2_ofs("solution.dat");
+      sol2_ofs.precision(8);
+      Vector      vals,coords;
+      GridFunction *nodes = mesh->GetNodes();
+      if (!nodes)
+      {
+         nodes = new GridFunction(fespace);
+         mesh->GetNodes(*nodes);
+      }
+
+      for (int i = 0; i <  mesh->GetNE(); i++)
+      {
+         int geom       = mesh->GetElementBaseGeometry(i);
+         RefinedGeometry *refined_geo = GlobGeometryRefiner.Refine(( Geometry::Type)geom,
+                                                                   lod, 1);
+
+         x.GetValues(i, refined_geo->RefPts, vals);
+         nodes->GetValues(i, refined_geo->RefPts, coords);
+
+         for (int j = 0; j < vals.Size(); j++)
```

Comment:
This output of solution values is better than no output, and it is not the fault of this PR that MFEM has no visualization tool yet for these 1D NURBS solutions. However, it is unfortunate that these solution values cannot be immediately visualized. Also, `solution.dat` has redundant points, at the endpoints of elements. A user needs to set up their own 1D visualization tool, which is not too difficult, but it would take a bit of work, and the redundant points need to be handled. Perhaps a brief comment would be helpful, to inform the user that there is currently no visualization tool in MFEM for this data, and that there are redundant points output at the element interfaces.

Reply:
Visualization  is very easy, as it is 1D. Gnuplot will do...

I cleaned the output. It is sorted and made unique. 
Perhaps those few lines could be moved to Gridfunction?